qiskit/compiler/transpile.py

```diff
@@ -299,18 +342,31 @@ def _parse_basis_gates(basis_gates, backend, circuits):
     return basis_gates
 
 
-def _parse_coupling_map(coupling_map, backend, num_circuits):
+def _parse_coupling_map(coupling_map, backend, circuits):
```

Comment:
why is this function changing at all in this PR? It's already a bit hard for me to follow what's going on, and this one seems completely unrelated. 

Same for `_parse_output_name()`

Reply:
Sure.. the goal was to encapsulate the cm checks here. But I can do that in a following PR. Undone in e1d7f10da for `_parse_coupling_map`.

For the `_parse_output_name` case, there was unnecessary logic there, because `None` means "keep the input name". Undone in 7ad1521f2.